devices/door_contact_sensor.py

```diff
@@ -5,4 +5,7 @@ def __init__(self, devices, alias, value_key):
         super().__init__(devices, alias, value_key, BooleanSensor.SENSOR_TYPE_DOOR_CONTACT)
     
     def get_numeric_value(self, value, device):
-        return 0 if value else 1
\ No newline at end of file
+        return 0 if value else 1
+        
+    def get_sn_values(self, key, value, device):
```

Comment:
Why do we need this method in addition to `get_numeric_value` and `get_string_value`? And anyway I guess we could move it to `Device` class and override where required to avoid code duplicity.

Reply:
I bundled the both values methods into a single version because I saw use for that when combining state and brightness. Plus, you always need to combine them in combination so why not make one function. But it's indeed a good idea to move the default merging function into the `Device` class